docs/function.rst

```diff
@@ -0,0 +1,147 @@
+.. _function:
+
+Function
+========
+
+Melody functions are user-written Python functions that perform the
```

Comment:
I think Melody Objective Function is the way to go. Once introduced you could just refer to it as the Objective Function. Once that's done then I'm happy.